File: apps/userdocs_web/assets/js/runner/lib/domain/annotation.ts

import { AnnotationType } from './annotation_type'

export interface Annotation {
  id: string,
  name: string,
  label: string,
  x_orientation: string,
  y_orientation: string,
  size: number,
  color: string,
  thickness: number,
  x_offset: number,
  y_offset: number,
  font_size: number,
  font_color: string,
  annotation_type: AnnotationType
}

File: apps/userdocs_web/lib/userdocs_web/api/schemas/annotation.ex
defmodule UserDocsWeb.API.Schema.Annotation do
  use Absinthe.Schema.Notation
  alias UserDocsWeb.API.Resolvers

  object :annotation do
    field :id, :id
    field :name, :string
    field :label, :string
    field :x_orientation, :string
    field :y_orientation, :string
    field :size, :integer
    field :color, :string
    field :thickness, :integer
    field :x_offset, :integer
    field :y_offset, :integer
    field :font_size, :integer
    field :font_color, :string

    field :annotation_type, :annotation_type, resolve: &Resolvers.AnnotationType.get_annotation_type!/3
  end
end
